pgpubsub/listen.py

```diff
@@ -21,11 +23,50 @@
 POLL = True
 
 
+def start_listen_in_a_process(
+    channels: Union[List[BaseChannel], List[str]] = None,
+    recover: bool = False,
+    autorestart_on_failure: bool = True,
+    start_method: str = 'spawn',
+    name: Optional[str] = None,
+) -> multiprocessing.Process:
+    connection.close()
+    multiprocessing.set_start_method(start_method, force=True)
+    logger.info('Restarting process')
+    if channels:
+        channels = [c if isinstance(str, c) else c.name() for c in channels]
+    if start_method == 'fork':
+        logger.debug('  using fork')
+        process = multiprocessing.Process(
+            name=name,
+            target=listen,
+            args=(channels, recover, autorestart_on_failure, 'fork'),
+        )
+    elif start_method == 'spawn':
+        args = [sys.argv[0], 'listen', '--worker', '--worker-start-method', 'spawn']
+        if recover:
+            args.append('--recover')
+        if not autorestart_on_failure:
+            args.append('--no-restart-on-failure')
+        if channels:
+            args.extend(channels)
+        logger.debug(f'  with {args=}')
+        process = multiprocessing.Process(
+            name=name, target=execute_from_command_line, args=(args, )
+        )
+    else:
+        raise ValueError(f'Unsupported start method {start_method}')
+
+    process.start()
+    return process
+
+
 def listen(
     channels: Union[List[BaseChannel], List[str]] = None,
     recover: bool = False,
+    autorestart_on_failure: bool = True,
+    start_method: str = 'spawn',
 ):
-    multiprocessing.set_start_method('fork', force=True)
```

Comment:
When I look at the commit history, I'm reminded that I added this to work around https://code.djangoproject.com/ticket/31169. I can't fully remember how I was getting that error/how it occurred. Hopefully it isn't still an issue.

Reply:
There is definitely no problem on mac when pytest is used to run tests - I am using mac and tests pass for all these tox envs

```
  38-django22: commands succeeded
  38-django32: commands succeeded
  38-django40: commands succeeded
  39-django22: commands succeeded
  39-django32: commands succeeded
  39-django40: commands succeeded
  310-django22: commands succeeded
  310-django32: commands succeeded
  310-django40: commands succeeded
```

I would like to figure out if the parallel execution via `manage.py test` has this issue. The command does not find any tests for me when run on my host. I'm trying now to use the new docker container to run tests via `manage.py test`.

Reply:
```
$ docker compose exec django-pgpubsub-app ./manage.py test
```
does not work btw. Here's the error:
```
service "django-pgpubsub-app" is not running container #1
```
The fix https://github.com/docker/compose/issues/10761#issuecomment-1729322135 is to use service name:

```
docker compose exec app ./manage.py test
```
But in this case I have the same issue as locally - namely no tests are found:
```
$ docker compose exec app ./manage.py test
System check identified no issues (0 silenced).

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
```

Reply:
Thanks for noticing that. 
docker compose exec app pytest is actually the correct one to use - i've fixed that on master.

Reply:
pytest works fine and even tox (which uses pytest as well) works.